### PR TITLE
Initial tests for writing functions

### PR DESCRIPTION
**Pull request by CalCraven**
This PR aims to add tests for all of the utility readers and writers used in the repo. To aid in this testing, a set of files are all stored in `tests/files/` for ease of access and testing on.

- [ ] Add files output from gaussian
- [ ] Fix a few typos for guass used instead of gauss
- [ ] Finish tests that are just set to pass for now

### CODE REVIEW

**Comment by bc118** [comment]
Notes: HEATOM is not supported in mosdef-GOMC writers, it always writes ATOM in pdb files

File: mosdef_dihedral_fit/utils/file_read_and_write.py
Comment (bc118) [inline mosdef_dihedral_fit/utils/file_read_and_write.py:1]: ATOM is currently the only output for mosdef-GOMC pdb files.

**Comment by bc118** [comment]
Should we put all these test files in the utils/files (https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/tree/main/mosdef_dihedral_fit/utils/files), where all the other ones are?   

Maybe I don’t understand the reason.

**Comment by CalCraven** [comment]
> Should we put all these test files in the utils/files (https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/tree/main/mosdef_dihedral_fit/utils/files), where all the other ones are?

In terms of packaging, typically you want files that are solely there for test validation to exist in the /tests module. /utils/files would be something more like file templates or something that is actually used during the course of operating the package.

**Comment by CalCraven** [comment]
Looks like macos and ubuntu tests are giving different outputs for the fifth element of the dihedral angle (macos=+180.0, ubuntu=-180.0). Probably something with the get_matching_dihedral_info_and_opls_fitting_data 

Looks like maybe a bug in this [line](https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/blob/21ed28a2b5361097687c080c814eb430feb2fb89/mosdef_dihedral_fit/utils/math_operations.py#L212):
```
  if dihedral_angle_degrees == 180:
      dihedral_angle_degrees == -180
```
Should be?
```
  if dihedral_angle_degrees == 180:
      dihedral_angle_degrees = -180
```
Regardless, I'm guessing the difference is coming from a dot or cross product direction.

**Comment by bc118** [comment]
> Looks like macos and ubuntu tests are giving different outputs for the fifth element of the dihedral angle (macos=+180.0, ubuntu=-180.0). Probably something with the get_matching_dihedral_info_and_opls_fitting_data
> 
> Looks like maybe a bug in this [line](https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/blob/21ed28a2b5361097687c080c814eb430feb2fb89/mosdef_dihedral_fit/utils/math_operations.py#L212):
> 
> ```
>   if dihedral_angle_degrees == 180:
>       dihedral_angle_degrees == -180
> ```
> 
> Should be?
> 
> ```
>   if dihedral_angle_degrees == 180:
>       dihedral_angle_degrees = -180
> ```
> 
> Regardless, I'm guessing the difference is coming from a dot or cross product direction.

Yeah, I agree, you are likely correct here

**Comment by CalCraven** [comment]
> ATOM is currently the only output for mosdef-GOMC pdb files.

That is fine, I think we can support HETATM since that's what you get from MoSDeF pdb writers and as far as I can tell the data can get parsed the same way.

**Review by bc118 — APPROVED** [review]
LGTM


Since we created the same files again in "test/files" (used for unit tests).  Let's remove all the files from "utils/files/" and move them to "test/files", in a quick next PR.  This way all the test files are in 1 location.....

**Pull request merged**